snuba/migrations/group_loader.py

```diff
@@ -0,0 +1,308 @@
+from __future__ import annotations
+
+from abc import ABC, abstractmethod
+from importlib import import_module
+from typing import Sequence
+
+from snuba.migrations.errors import MigrationDoesNotExist
+from snuba.migrations.migration import Migration
+
+
+class GroupLoader(ABC):
+    """
+    Provides the list of migrations associated with an group and a loader that returns
+    the requested migration.
+    """
+
+    @abstractmethod
+    def get_migrations(self) -> Sequence[str]:
+        """
+        Returns the list of migration IDs in the order they should be executed.
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def load_migration(self, migration_id: str) -> Migration:
+        raise NotImplementedError
+
+
+class DirectoryLoader(GroupLoader, ABC):
+    """
+    Loads migrations that are defined as files of a directory. The file name
+    represents the migration ID.
+    """
+
+    def __init__(self, module: str) -> None:
```

Comment:
is this a `module_name` or `module_path`?

Reply:
Updated to `module_path`